wrappers/desktop.go

```diff
@@ -171,6 +171,11 @@ func sanitizeDesktopFile(s *snap.Info, desktopFile string, rawcontent []byte) []
 		newContent.Grow(len(bline) + 1)
 		newContent.Write(bline)
 		newContent.WriteByte('\n')
+
+		// insert snap name
+		if bytes.Equal(bline, []byte("[Desktop Entry]")) {
+			newContent.Write([]byte("X-SnapName=" + s.SnapName() + "\n"))
```

Comment:
This should probably be `InstanceName` rather than `SnapName`.  For the purpose of matching against the package ID xdg-desktop-portal determines from the AppArmor security label, I think the instance name is what we want.